src/features/linkApplicationToCustomer/LinkApplicationToCustomerContainer.tsx

```diff
@@ -90,19 +113,66 @@ const LinkApplicationToCustomerContainer = ({
   const filteredCustomersData = getFilteredCustomersData(customersData);
 
   const handleCreateCustomer = () => {
-    const { firstName, lastName, address, email, phoneNumber, zipCode, municipality } = application;
+    const {
+      firstName,
+      lastName,
+      address,
+      email,
+      phoneNumber,
+      zipCode,
+      municipality,
+      businessId,
+      companyName,
+    } = application;
 
-    createNewCustomer({
-      variables: {
-        firstName: firstName,
-        lastName: lastName,
-        address: address,
-        postalCode: zipCode,
-        city: municipality,
-        phone: phoneNumber,
-        email: email,
-      },
-    });
+    if (businessId === '') {
+      createNewCustomer({
+        variables: {
+          firstName: firstName,
+          lastName: lastName,
+          addresses: [
+            { address, postalCode: zipCode, city: municipality, primary: true, addressType: AddressType.NONE },
+          ],
+          phone: phoneNumber,
+          email: email,
+        },
+      }).then(({ data }) => {
+        if (!data?.createProfile?.profile?.id) {
+          return;
+        }
+        return createNewBerthProfile({
+          variables: { input: { id: data.createProfile.profile.id } },
+        });
+      });
+    } else {
+      createNewCustomer({
+        variables: {
+          firstName: firstName,
+          lastName: lastName,
+          phone: phoneNumber,
+          email: email,
+        },
+      }).then(({ data }) => {
+        if (!data?.createProfile?.profile?.id) {
+          return;
+        }
+        return createNewBerthProfile({
+          variables: {
+            input: {
+              id: data.createProfile.profile.id,
+              organization: {
+                businessId,
+                name: companyName,
+                organizationType: OrganizationType.COMPANY,
+                address,
+                postalCode: zipCode,
+                city: municipality,
+              },
+            },
+          },
+        });
+      });
+    }
```

Comment:
One optional note, the same logic might be used in other pages for creating a customer, so we might consider extracting this function now or later to some external module.

Reply:
Done.